File: .yalc/@entur/atb-mobile-client-sdk/challenge/bluetooth/model/bluetooth-model-ios-lib/Sources/bluetooth-model-ios-lib/header/ChallengeAdvertisingFrameGroupHeader.swift

import Foundation

// Header with the fields common to a frame group (from the same Challenge).

public class ChallengeAdvertisingFrameGroupHeader : Hashable, CustomStringConvertible {
 
    // implementation note: this could have been a subclass of the header
    // but we need the header as a map key, so keep it as is
    
    let clientId : UInt32
    let correlationId : UInt32
    let channel : UInt32
    
    public var description: String {
        return "clientId: \(clientId) correlationId: \(correlationId) channel: \(channel)"
    }

    convenience init(h : ChallengeAdvertisingFrameHeader) {
        self.init(clientId: h.clientId, correlationId: h.correlationId, channel: h.channel)
    }

    init(clientId: UInt32, correlationId: UInt32, channel : UInt32) {
        self.clientId = clientId;
        self.correlationId = correlationId;
        self.channel = channel;
    }
    
    public func hash(into hasher: inout Hasher) {
        hasher.combine(clientId)
        hasher.combine(correlationId)
        hasher.combine(channel)
    }

    public static func == (a:ChallengeAdvertisingFrameGroupHeader, b:ChallengeAdvertisingFrameGroupHeader) -> Bool {
        return a.clientId == b.clientId && a.correlationId == b.correlationId && a.channel == b.channel
    }
    
}

File: .yalc/@entur/atb-mobile-client-sdk/challenge/bluetooth/model/bluetooth-model-java-lib/src/main/java/no/entur/abt/android/bluetooth/header/ChallengeAdvertisingFrameGroupHeader.java
package no.entur.abt.android.bluetooth.header;

import java.util.Objects;

/**
 *
 * Header with the fields common to a frame group (from the same Challenge).
 *
 */

public class ChallengeAdvertisingFrameGroupHeader {

	protected final int clientId;
	protected final int correlationId;
	protected final int channel;
	protected final int hash;

	public ChallengeAdvertisingFrameGroupHeader(ChallengeAdvertisingFrameHeader h) {
		this(h.getClientId(), h.getCorrelationId(), h.getChannel());
	}

	public ChallengeAdvertisingFrameGroupHeader(int clientId, int correlationId, int channel) {
		this(clientId, correlationId, channel, Objects.hash(clientId, correlationId, channel));
	}

	public ChallengeAdvertisingFrameGroupHeader(int clientId, int correlationId, int channel, int hash) {
		this.clientId = clientId;
		this.correlationId = correlationId;
		this.channel = channel;
		this.hash = hash;
	}

	@Override
	public boolean equals(Object o) {
		if (this == o) {
			return true;
		}
		if (o instanceof ChallengeAdvertisingFrameGroupHeader) {
			ChallengeAdvertisingFrameGroupHeader frameKey = (ChallengeAdvertisingFrameGroupHeader) o;
			return clientId == frameKey.clientId && correlationId == frameKey.correlationId && channel == frameKey.channel;
		}
		return false;
	}

	@Override
	public int hashCode() {
		return hash;
	}

	@Override
	public String toString() {
		return "FrameKey{" + "clientId=" + clientId + ", correlationId=" + correlationId + ", channel=" + channel + '}';
	}

	public int getCorrelationId() {
		return correlationId;
	}

	public int getClientId() {
		return clientId;
	}

	public int getChannel() {
		return channel;
	}

}
